hpy/debug/src/_debugmod.c

```diff
@@ -0,0 +1,212 @@
+// Python-level interface for the _debug module. Written in HPy itself, the
+// idea is that it should be reusable by other implementations
+
+// NOTE: hpy.debug._debug is loaded using the UNIVERSAL ctx. To make it
+// clearer, we will use "uctx" and "dctx" to distinguish them.
+
+#include "hpy.h"
+#include "debug_internal.h"
+
+static UHPy new_DebugHandleObj(HPyContext uctx, UHPy u_DebugHandleType,
+                               DebugHandle *handle);
+
+
+HPyDef_METH(new_generation, "new_generation", new_generation_impl, HPyFunc_NOARGS)
+static UHPy new_generation_impl(HPyContext uctx, UHPy self)
+{
+    HPyContext dctx = hpy_debug_get_ctx(uctx);
```

Comment:
Obtaining a new debug ctx from the `uctx` requires that the `uctx` be the same in this module as in the module being debugged, yes? Which means there essentially needs to be only one `ctx` per interpreter? This is already true in the CPython and PyPy (I think) implementations, but it's not currently *required* by HPy.

I'm not quite sure how to remove this constraint right now -- the ctx is fairly invisible from Python code, and that likely needs to stay that way.

Reply:
I tried to explain more what are the implicit and explicit assumptions in commit 6bad6c3. If it's still unclear, please comment/review/reword directly on PR #164